src/stats/serialize.rs

```diff
@@ -0,0 +1,21 @@
+pub mod instant_to_epoch_ms {
+    use serde::{Serialize, Serializer};
+    use std::time::{SystemTime, UNIX_EPOCH};
+    use tokio::time::Instant;
+
+    pub fn serialize<S>(instant: &Instant, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let system_now = SystemTime::now();
+        let instant_now = Instant::now();
+        let approx = system_now - (instant_now - *instant);
+        let epoch = approx
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards");
+
+        let epoch_ms = epoch.as_millis() as f64 / 1000.0;
```

Comment:
This could definitely use a test, and perhaps some clarification of the code. The output of this is an approximate timestamp as a `f64`, with seconds to the left of the decimal point and milliseconds to the right.

Reply:
In trying to write a test, the fact that the data sources are all `Instant`s makes things pretty difficult. There doesn't seem to be any way in the language to convert between instants and real time, and the docs go out of their way to say that instants cannot be compared to time.

The underlying data collected in stats is all `Instant`s, which is why I modeled stats that way. A longer term refactor might be to find the locations where those are being saved and convert them to `SystemTime`s or another data structure more closely connected to real world time... something where a specific instant in time can be hard coded into a test, and be serialized to a specific float.

Reply:
If we are only measuring elapsed time then `Instant` are exactly the right choice, that's what they are for after all. 

> This could definitely use a test, and perhaps some clarification of the code. The output of this is an approximate timestamp as a f64, with seconds to the left of the decimal point and milliseconds to the right.

Can we clarify this with a comment or rename the variable `epoch_seconds`?

Reply:
That sounds great. Just pushed a change to rename to `instant_to_epoch_seconds`.